### PR TITLE
Support for AWS Instance Metadata Service v2

### PR DESCRIPTION
**Pull request by joshathysolate**
### Description
This PR corresponds to PR 270 in the smallstep/certificates repo in order to add a new flag called --imds-version so that customers can specify which version(s) of the IMDS they want to support on their AWS provisioner.

By default existing AWS provisioners and any new provisioner where --imds-version is not specified will default to V2 first and then falling back to V1 if an error or an HTTP status of 400 or greater is returned by the V2 service.

### CODE REVIEW

File: command/ca/provisioner/add.go [hunk 1]
```diff
@@ -491,12 +498,29 @@ func addAWSProvisioner(ctx *cli.Context, name string, provMap map[string]bool) (
 		return nil, err
 	}
 
+	// validate IMDS versions before we send them over to the CA
+	imdsVersions := ctx.StringSlice("imds-version")
+	if len(imdsVersions) == 0 {
+		imdsVersions = []string{"v2", "v1"}
+	}
```
Comment (maraino) [inline command/ca/provisioner/add.go:505]: You can set the defaults in the `cli.StringSliceFlag.Value` property.

Comment (maraino) [reply command/ca/provisioner/add.go:505]: You can also leave this as empty and relay in the default logic of the CA.

**Comment by maraino** [comment]
Requires smallstep/certificates#270

**Comment by jokreliable** [comment]
does https://github.com/smallstep/certificates/pull/683 make this unnecessary? (seems related)

**Comment by maraino** [comment]
Yes @jokreliable this is not necessary anymore. Thanks for noticing.

**Pull request closed without merging**